README.md

```diff
@@ -11,34 +11,53 @@ EvaP is the course evaluation system used internally at Hasso Plattner Institute
 
 For the documentation, please see our [wiki](https://github.com/e-valuation/EvaP/wiki).
 
+## Development Setup (Podman)
 
-## Installation (for Development)
+To develop EvaP, we recommend that you set up a development container.
+This can be done with the configuration file provided by us.
+You will have to install [`git`](https://git-scm.com/downloads) and [`podman`](https://podman.io/).
+Clone the EvaP repository and create the container using
+```bash
+git clone --recurse-submodules https://github.com/e-valuation/EvaP.git
+cd EvaP
+podman build --tag evap-image nix/
+podman create --name evap-container --userns=keep-id:uid=1001,gid=1001 --volume $PWD:/evap --publish 8000:8000 evap-image
+```
 
-The easiest setup using [Vagrant](https://www.vagrantup.com) is shown here.
+From now on, you can use the created container whenever you want to work on EvaP.
+Start the container with `podman start evap-container` and enter it with `podman exec -it evap-container machinectl shell -q evap@`.
+When entering the container for the first time, it may take a while until the database and development environment have finished setting up.
+Once you see a shell prompt like `evap@5b17c1be4315:/evap$`, you should be ready.
+To run the final initialization steps, run the command `initialize-setup`.
+Now, you can start EvaP by running `./manage.py run`.
+Open your web browser at http://localhost:8000/ and login with email `evap@institution.example.com` and password `evap`.
+You can exit the container by pressing `Ctrl-D` and then stop it with `podman stop evap-container` (or it will stop automatically when you shut down your computer).
```

Comment:
Sledgehammer approach that I think would be totally fair:

profide some `nix run .#dev-shell` (or whatever) command that spawns a new bash and inside that bash
* loads autocompletions
* runs whatever nix commands builds the envinroment (`nix build`?) -- assuming that this finished in <0.1s when already done.
* checks if `initialize-setup` has been run, and if not, runs it (check could be: localsettings exists, or npm if everything is fine, or any other side effect from `initialize-setup`)
* starts `nix run .#services --tui` in the background -- assuming that now, everything is "healthy" in less than a second.

That way, people essentially get the bash that they would have got with Vagrant. Yes, it's a bash, and not the shell of your choice, but for >90% of contributors, it gets the job done.

I hope nix process-compose can do something like that.

Reply:
With `shellHook` from `pkgs.mkShell`, we can essentially do anything we want, so I think it wouldn't be a `nix run` but just a richer `nix develop`.

Process compose also has up and down commands (`nix run .#services -- up --detached`, `nix run .#services -- down`), this hides the output but let's us inspect it with `nix run .#services -- attach`. We can also recommend that to everyone, if we are really against running a separate terminal window. Starting the DB with a shell hook seems counterintuitive to me, because it is not stopped again when the shell closes (and I don't think there is a reliable way to do so).

If you really insist, we can automatically do `initialize-setup` as part of the shell hook. I am hesitant to do so, because it seems very fragile and counter-intuitive. I think that we can, some time in the future, have the steps of `initialize-setup` built into the process compose workflow automatically, I just haven't gotten into how to do this yet. I would first tackle the node dependencies I think

Reply:
> (and I don't think there is a reliable way to do so).

I think [using `trap`](https://stackoverflow.com/a/2173421/12345551) would suffice for us. I'd also be fine with a wrapper script that is essentially 1. services up, 2. run an interactive bash, 3. services down.

> I am hesitant to do so, because it seems very fragile and counter-intuitive

If it's a shell hook, yes. This is why I proposed a dedicated command, with the intention of being newb-friendly. I just intend to provide the "run this command to get a shell with everything setup"-experience.

> I would first tackle the node dependencies I think

(Not as important to me, I'm fine with any complexity that experienced people can navigate around, I just want to provide a good experience to new devs)

Reply:
I moved all of `initialize-setup` into process compose jobs now, so the workflow would be `nix run .#services-full -- --detached` and `nix run .#services-full -- down` paired with `nix develop`. It's basically the same workflow as `vagrant up`, `vagrant halt` and `vagrant ssh`.

Since I feel that this workflow is okay, time is running out, and I am not too confident setting up everything with `trap`, I would prefer to not provide a do-everything script at this time and instead think about this in a follow up.

Reply:
Works for me, thanks :)